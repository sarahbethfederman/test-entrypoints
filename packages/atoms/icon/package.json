{
  "name": "@auscred/icon",
  "version": "0.0.1",
  "source": "src/Icon.tsx",
  "main": "dist/Icon.js",
  "types": "dist/Icon.d.ts",
  "directories": {
    "lib": "lib"
  },
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "devDependencies": {
    "jest": "^23.1.0",
    "@lui/story-utils": "^1.0.0",
    "@storybook/addon-knobs": "^3.4.8",
    "@types/storybook__addon-knobs": "^3.3.1"
  },
  "scripts": {
    "build": "node ./src/build.js && tsc -p ./tsconfig.json"
  },
  "dependencies": {
    "svg-sprite": "^1.4.0"
  }
}
