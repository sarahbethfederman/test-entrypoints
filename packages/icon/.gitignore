src/icons-compiled/*
src/index.ts