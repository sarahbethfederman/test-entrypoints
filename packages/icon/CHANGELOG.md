## [9.6.2] - 2020-09-02 - [SUB-609][https://creditandfinance.atlassian.net/browse/sub-609]

## 10.0.0-beta.0

### Minor Changes

- preconstruct infra update

### Patch Changes

- Updated dependencies [undefined]
  - @lendi-ui/color@6.0.0-beta.0
  - @lendi-ui/theme@10.0.0-beta.0

### Added

- Added the domain bot Icon && Updated the autobot Icon name to LendiBotIcon

## [9.5.2] - 2020-08-27 - [SUB-601][https://creditandfinance.atlassian.net/browse/sub-601]

### Updated

- Updated Domain Home Loans for Team View

## [9.5.1] - 2020-08-18 - [SUB-461][https://creditandfinance.atlassian.net/browse/sub-461]

### Added

- Added Domain Home Loans logos

## [9.5.0] - 2020-04-28 - [SUB-242][https://creditandfinance.atlassian.net/browse/sub-242]

### Added

- Added autobot icon

## [9.4.1] - 2020-04-28 - [SUB-hotfix-unreadMessageClass]

### Added

- Updated the fillPath class on the unread message icon.

## [9.4.0] - 2020-04-28 - [SUB-hotfix-redDot]

### Added

- Added red dot icon.

## [9.3.0] - 2020-03-18 - [HUB-hotfix-domainIcon](https://creditandfinance.atlassian.net/browse/FUNNEL-1081)

### Added

- Added domain logomark icon.
- Added domain teamview logo icon.

## [9.2.0] - 2020-02-19 - [HUB-893](https://creditandfinance.atlassian.net/browse/HUB-893)

### Added

- Added unread icon.

## [9.1.1] - 2019-11-22

### Changed

- Fixed the colours in the new PropertyReport icon.

## [9.1.0] - 2019-11-22 - [HUB-664](https://creditandfinance.atlassian.net/browse/HUB-664)

### Added

- Added in relation to: HUB-664, HUB-665 and HUB-695
- Adds: ArrowRightCircle, BullseyeArrow, LightbulbOn, HomeMapMarker, HomeAccount, PropertyReport, CalculatorVariant, IcAssignment and CalendarMonth icons.

## [9.0.2] - 2019-09-27

- Change color to an optional property.

## [9.0.1] - 2019-09-26 [Release Notes](https://creditandfinance.atlassian.net/wiki/spaces/HUB/pages/803930391/Upcoming+Major+Changes)

- Bumped lendi-ui dependencies to latest versions

## [9.0.0] - 2019-9-23 [Release notes](https://creditandfinance.atlassian.net/wiki/spaces/HUB/pages/803930391/Upcoming+Major+Changes)

### Changed

- Uses Theme v7
- Update typescript to version 3.5.2 and react to version 16.8.4 [HUB-151](https://creditandfinance.atlassian.net/browse/HUB-151)

## [8.2.0] - 2019-07-16 - [HUB-330](https://creditandfinance.atlassian.net/browse/HUB-330)

### Changed

- Change icon default color to 'currentColor'

## [8.1.0] - 2019-06-07 - [HUB-302](https://creditandfinance.atlassian.net/browse/HUB-302)

### Changed

- Added the vertical-split icon

## [8.0.1] - 2019-06-07 - [HUB-264](https://creditandfinance.atlassian.net/browse/HUB-264)

### Changed

- Fix the icon alignment issue.

## [8.0.0] - 2019-05-29 - [HUB-144](https://creditandfinance.atlassian.net/browse/HUB-144)

### Deleted

- FileDownload, SuccessTwo, Warn, Error, Mobile (duplicate icons)
- All fancy icons (moved to the new fancy-icon package)

## Changed

- Renamed icons to match material.ui.

## [7.2.0] - 2019-05-17 - [HUB-174](https://creditandfinance.atlassian.net/browse/HUB-174)

### Changed

- Module format reverted back to CJS.

### Deleted

- Dropped "module" property from package.json.

## [7.1.2] - 2019-05-09 - [HUB-92](https://creditandfinance.atlassian.net/browse/HUB-92)

### Changed

- Updated CHANGELOG to match required parsing format.

## [7.1.1] - 2019-05-08 - [HUB-54](https://creditandfinance.atlassian.net/browse/HUB-54)

### Changed

- Change module main entry point to esm in package.json

## [7.1.0] - 2019-05-02 - [HUB-126](https://creditandfinance.atlassian.net/browse/HUB-126)

### Added

- Added the following new icons: file-upload & FileDownload as an alias to GetApp.

## [7.0.1] - 2019-05-02 - [HUB-132](https://creditandfinance.atlassian.net/browse/HUB-132)

### Changed

- Fix MS's Edge Icon rendering issue

## [7.0.0] - 2019-04-30 - [HUB-54](https://creditandfinance.atlassian.net/browse/HUB-54)

### Deleted

- Remove the CJS bundles.

## [6.0.0] - 2019-04-16 - [HUB-49](https://creditandfinance.atlassian.net/browse/HUB-49)

### Changed

- Remove Left and Right Icons.

## [5.3.1] - 2019-04-08 - [HUB-71](https://creditandfinance.atlassian.net/browse/HUB-71)

### Changed

- Added styling to the icon container to align text and icon baseline when using inline icons.

## [5.3.0] - 2019-04-01 - [HUB-53](https://creditandfinance.atlassian.net/browse/HUB-53)

### Added

- Added the following new icons: attach-file, assignment-returned, autorenew, assessment, save, credit-card, assignment-return, people, more-horiz, done-all, assignment-turned-in, error2, search, date-range, assignment, open-with, dashboard, timeline, gavel, trending-flat, report-problem, trending-down, delete, fullscreen, delete-outline, remove-circle, sentiment-satisfied-alt, launch, edit, add-alert, trending-up, help, history, schedule, room, notification-important, get-app, share, highlight-off, more-vert, favorite-border, hourglass-empty, info2, monetization-on, thumb-up, help-outline, error-outline, add, check2, warning, build, computer, data-usage, alarm, description, stay-current-portrait, power-settings-new, account-balance

## [5.2.0] - 2019-03-15 - [BRAVO-576](https://creditandfinance.atlassian.net/browse/BRAVO-576)

### Added

- Added un-circled warn, info and success icons `info-two success-two warn-two`

## [5.1.0] - 2019-03-06 - [BRAVO-572](https://creditandfinance.atlassian.net/browse/BRAVO-572)

### Added

- Added Maintenance icon and PageNotFound icon.

## [5.0.1] - 2019-03-05 - [BRAVO-387](https://creditandfinance.atlassian.net/browse/BRAVO-387)

### Added

- Added Domain logo.

## [5.0.0]

### Changed

- Breaking change version update

## [4.4.0]

### Changed

- Bump version for left and right icons
