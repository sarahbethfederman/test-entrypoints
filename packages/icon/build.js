const SVGSpriter = require('svg-sprite');
const fs = require('fs');
const path = require('path');
const mkdirp = require('mkdirp');

const directory = `${__dirname}/src/icons`;

const spritify = new SVGSpriter({
  dest: 'dist',
  mode: {
    symbol: true,
    inline: true,
  },
});

fs.readdir(directory, (err, files) => {
  if (err) {
    console.error(err);
    return;
  }

  files.forEach((file) => {
    if (file.endsWith('.svg')) {
      spritify.add(
        path.resolve(`${directory}/${file}`),
        file,
        fs.readFileSync(`${directory}/${file}`, { encoding: 'utf-8' })
      );
    }
  });

  spritify.compile(function(error, result, data) {
    for (var mode in result) {
      for (var resource in result[mode]) {
        fs.writeFileSync(`../../website/static/sprite.symbol.svg`, result[mode][resource].contents);
      }
    }
  });
});
