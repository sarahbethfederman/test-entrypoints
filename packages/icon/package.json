{
  "name": "@lendi-ui/icon",
  "version": "9.3.0",
  "main": "dist/cjs/index.js",
  "source": "src/index.tsx",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist"
  ],
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "peerDependencies": {
    "@lendi-ui/theme": "^8.0.0",
    "@types/react": "^16.8",
    "@types/styled-components": "^4.1.18",
    "react": "^16.8",
    "styled-components": "^4"
  },
  "devDependencies": {
    "@svgr/core": "^2.2.0",
    "@lendi-ui/typography": "^4.0.0",
    "@lendi-ui/spacing": "^6.0.0"
  },
  "dependencies": {
    "@lendi-ui/color": "^5.0.0"
  },
  "scripts": {
    "build:svg": "node ./build/index.js",
    "build:cjs": "tsc --project tsconfig.cjs.json",
    "build:types": "tsc --project tsconfig.types.json",
    "build": "yarn run build:svg && yarn run build:types && yarn run build:cjs"
  }
}
