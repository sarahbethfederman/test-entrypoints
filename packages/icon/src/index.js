import React from 'react';

const Icon = () => <span>Icon</span>;

export default Icon;
