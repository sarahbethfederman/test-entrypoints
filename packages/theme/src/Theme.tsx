import * as React from 'react';
import styled, { ThemeProvider, injectGlobal } from 'styled-components';
import { theme } from './theme/lendi';
import { ThemeMap } from './types';

interface ThemeProps {
  kind?: ThemeMap;
  children?: React.ReactNode;
}

// Any cross browser resets should occur here
// tslint:disable-next-line no-unused-expression
injectGlobal`
  html, body {
    margin: 0;
    padding: 0;
  }

  :root {
    --lendi-ui-size: 14px;
  }

  @media (min-width: 36.0625em) {
    :root {
      --lendi-ui-size: 16px;
    }
  }
`;

const Root = styled.div`
  color: ${theme.colors.shade[700]};
`;

export function Theme(props: ThemeProps) {
  const { children } = props;
  const kind = props.kind || theme;

  return (
    <ThemeProvider theme={kind}>
      <Root>{children}</Root>
    </ThemeProvider>
  );
}
