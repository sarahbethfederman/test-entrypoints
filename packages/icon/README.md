# @lendi-ui/icon

## Adding a new icon

Add the SVG into src/icons. Be thoughtful with naming, as this will determine the name of the component. For example, the file `computer-bank` will create a SVG component called `computerBank` (note that the kebab-case filename will become a camelCase component).

To add a new alias, navigate to the ./build/alias.json file and add the key as the original file name (in pascal case) and give it an array with all the alias' to add.

When this package is built, an equivalent react component will be created in src/icons-compiled.

## Installation

```
yarn add @lendi-ui/icon
```

## Usage

```jsx
import { Lock } from '@lendi-ui/icon';

<Lock color="secondary.500" />;
```

## Fill colours not working?

In this case, you will need to add a `fillPath` class to the element in the SVG that needs to be recoloured, this can be determined by inspecting the svg in the browser and changing the fill colour until you find the correct element.

## Properties

```
<PropTable>
  <PropTable.Entry required name="color" type={`string`} description="The Icon color"/>
</PropTable>
```
