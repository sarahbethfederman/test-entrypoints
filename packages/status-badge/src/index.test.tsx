import * as React from 'react';
import { mount } from 'enzyme';
import Theme from '@lendi-ui/theme';
import StatusBadge from './index';
import { Wrapper, IconWrapper, ContentWrapper } from './index.style';
import { InfoTwo, Check, WarnTwo } from '@lendi-ui/icon';

let element, variant;

function render(props) {
  element = mount(
    <Theme>
      <StatusBadge variant="success" hasIcon={true} statusText="COMPLETED" {...props} />
    </Theme>
  );
}

describe('StatusBadge', () => {
  it('it should mount StatusBadge component without icon', () => {
    render({ icon: false });
    expect(element.find(StatusBadge)).toHaveLength(1);
    expect(element.find(Wrapper)).toHaveLength(1);
    expect(element.find(ContentWrapper)).toHaveLength(1);
    expect(element.find(StatusBadge)).toMatchSnapshot();
  });

  it('it should mount StatusBadge component with icon', () => {
    render({});
    expect(element.find(StatusBadge)).toHaveLength(1);
    expect(element.find(Wrapper)).toHaveLength(1);
    expect(element.find(IconWrapper)).toHaveLength(1);
    expect(element.find(Check)).toHaveLength(1);
    expect(element.find(Check).props().color).toEqual('success.500');
    expect(element.find(ContentWrapper)).toHaveLength(1);
    expect(element.find(StatusBadge)).toMatchSnapshot();
  });

  it("it should mount the StatusBadge component when variant is 'error'", () => {
    variant = 'error';
    render({ variant });
    expect(element.find(WarnTwo)).toHaveLength(1);
    expect(element.find(WarnTwo).props().color).toEqual('error.500');
  });

  it("it should mount the StatusBadge component when variant is 'info'", () => {
    variant = 'info';
    render({ variant });
    expect(element.find(InfoTwo)).toHaveLength(1);
    expect(element.find(InfoTwo).props().color).toEqual('info.500');
  });

  it("it should mount the StatusBadge component when variant is 'success'", () => {
    variant = 'success';
    render({ variant });
    expect(element.find(Check)).toHaveLength(1);
    expect(element.find(Check).props().color).toEqual('success.500');
  });

  it("it should mount the StatusBadge component when variant is 'warn'", () => {
    variant = 'warn';
    render({ variant });
    expect(element.find(WarnTwo)).toHaveLength(1);
    expect(element.find(WarnTwo).props().color).toEqual('warn.500');
  });

  describe('test native props and Standard HTML Attributes', () => {
    it('should mount with Aria attributes', () => {
      const ARIA_LABEL = 'testLabel';
      const ARIA_DESCRIBE_BY = 'info';
      render({
        'aria-label': ARIA_LABEL,
        'aria-describedby': ARIA_DESCRIBE_BY,
      });
      const attributes = element.find(StatusBadge).props();
      expect(attributes['aria-label']).toBe(ARIA_LABEL);
      expect(attributes['aria-describedby']).toBe(ARIA_DESCRIBE_BY);
    });
    it('should mount with native props like id, tabIndex', () => {
      const TEXT_ID = 'testId';
      render({
        id: TEXT_ID,
        tabIndex: 1,
      });
      const attributes = element.find(StatusBadge).props();
      expect(attributes.id).toBe(TEXT_ID);
      expect(attributes.tabIndex).toBe(1);
    });
  });
});
