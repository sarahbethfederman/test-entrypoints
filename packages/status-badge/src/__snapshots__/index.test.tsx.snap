// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBadge it should mount StatusBadge component with icon 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  border-radius: 12px;
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  padding-right: calc(0.25 * var(--lendi-ui-size));
  border: 1px solid #4ec000;
  color: #4ec000;
  background-color: #f6fbf2;
}

.c1 {
  line-height: 0;
}

.c3 {
  font-size: calc(0.75 * var(--lendi-ui-size));
  line-height: calc(16 / 12);
  font-family: 'Open Sans',sans-serif;
}

.c2 {
  width: 1em;
  height: 1em;
  vertical-align: middle;
}

.c2 > svg {
  fill: #4ec000;
  width: inherit;
  height: inherit;
}

<StatusBadge
  hasIcon={true}
  statusText="COMPLETED"
  variant="success"
>
  <styled.div
    variant="success"
    withIcon={true}
  >
    <div
      className="c0"
    >
      <styled.span>
        <span
          className="c1"
        >
          <SuccessTwo
            color="success.500"
            height="1em"
          >
            <styled.span
              color="success.500"
              height="1em"
            >
              <span
                className="c2"
                color="success.500"
                height="1em"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <path
                    d="M18.265 6l-8.111 8.11-4.419-4.405L4 11.44l6.154 6.154L20 7.748z"
                    fillRule="evenodd"
                  />
                </svg>
              </span>
            </styled.span>
          </SuccessTwo>
        </span>
      </styled.span>
      <styled.span
        size="sm"
      >
        <span
          className="c3"
          size="sm"
        >
          COMPLETED
        </span>
      </styled.span>
    </div>
  </styled.div>
</StatusBadge>
`;

exports[`StatusBadge it should mount StatusBadge component without icon 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  border-radius: 12px;
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  padding-right: calc(0.25 * var(--lendi-ui-size));
  border: 1px solid #4ec000;
  color: #4ec000;
  background-color: #f6fbf2;
}

.c1 {
  line-height: 0;
}

.c3 {
  font-size: calc(0.75 * var(--lendi-ui-size));
  line-height: calc(16 / 12);
  font-family: 'Open Sans',sans-serif;
}

.c2 {
  width: 1em;
  height: 1em;
  vertical-align: middle;
}

.c2 > svg {
  fill: #4ec000;
  width: inherit;
  height: inherit;
}

<StatusBadge
  hasIcon={true}
  icon={false}
  statusText="COMPLETED"
  variant="success"
>
  <styled.div
    variant="success"
    withIcon={true}
  >
    <div
      className="c0"
    >
      <styled.span>
        <span
          className="c1"
        >
          <SuccessTwo
            color="success.500"
            height="1em"
          >
            <styled.span
              color="success.500"
              height="1em"
            >
              <span
                className="c2"
                color="success.500"
                height="1em"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                >
                  <path
                    d="M18.265 6l-8.111 8.11-4.419-4.405L4 11.44l6.154 6.154L20 7.748z"
                    fillRule="evenodd"
                  />
                </svg>
              </span>
            </styled.span>
          </SuccessTwo>
        </span>
      </styled.span>
      <styled.span
        size="sm"
      >
        <span
          className="c3"
          size="sm"
        >
          COMPLETED
        </span>
      </styled.span>
    </div>
  </styled.div>
</StatusBadge>
`;
